From 88a9ecaf6ee39c18326635e9ceebdb628102ce37 Mon Sep 17 00:00:00 2001
From: Witold Sadowski <wsadowski@marvell.com>
Date: Tue, 1 Jun 2021 06:04:16 -0700
Subject: [PATCH 1579/1921] drivers: spi: Fix mutex issue

Fix uninitialized mutex issue.

Signed-off-by: Witold Sadowski <wsadowski@marvell.com>
Change-Id: I0d948e4a2490a73ccfdee05056cd07b6c7e9ec62
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/53176
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/spi/spi-cadence-xspi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/spi/spi-cadence-xspi.c b/drivers/spi/spi-cadence-xspi.c
index fda1b3011158..e065cfdaab83 100644
--- a/drivers/spi/spi-cadence-xspi.c
+++ b/drivers/spi/spi-cadence-xspi.c
@@ -569,6 +569,8 @@ int cdns_xspi_configure(struct cdns_xspi_dev *cdns_xspi)
 		return ret;
 	}
 
+	mutex_init(&cdns_xspi->lock);
+
 	return ret;
 }
 
-- 
2.31.1

